openssl/src/pkcs7.rs

```diff
@@ -398,6 +419,45 @@ mod tests {
         assert!(content.is_none());
     }
 
+    #[test]
+    fn signers() {
+        let cert = include_bytes!("../test/cert.pem");
+        let cert = X509::from_pem(cert).unwrap();
+        let cert_digest = cert.digest(MessageDigest::sha256()).unwrap();
+        let certs = Stack::new().unwrap();
+        let message = "foo";
+        let flags = Pkcs7Flags::STREAM;
+        let pkey = include_bytes!("../test/key.pem");
+        let pkey = PKey::private_key_from_pem(pkey).unwrap();
+        let mut store_builder = X509StoreBuilder::new().expect("should succeed");
+
+        let root_ca = include_bytes!("../test/root-ca.pem");
+        let root_ca = X509::from_pem(root_ca).unwrap();
+        store_builder.add_cert(root_ca).expect("should succeed");
+
+        let pkcs7 =
+            Pkcs7::sign(&cert, &pkey, &certs, message.as_bytes(), flags).expect("should succeed");
+
+        let signed = pkcs7
+            .to_smime(message.as_bytes(), flags)
+            .expect("should succeed");
+
+        let (pkcs7_decoded, _) = Pkcs7::from_smime(signed.as_slice()).expect("should succeed");
+
+        let empty_certs = Stack::new().unwrap();
+        let signer_certs = pkcs7_decoded
+            .signers(&empty_certs, flags)
+            .expect("should succeed");
+        assert_eq!(empty_certs.len(), 0);
+        assert_eq!(signer_certs.len(), 1);
+        let signer_digest = signer_certs[0].digest(MessageDigest::sha256()).unwrap();
+        assert_eq!(*cert_digest, *signer_digest);
+        assert_ne!(
+            cert_digest.iter().fold(0u32, |acc, x| acc + u32::from(*x)),
```

Comment:
What's the purpose of this check?

Reply:
This was me proving to myself in the test that I actually got something likely to be a digest, and not all zeros. I can remove it if it's too much.

Reply:
Yeah, let's remove that since it's more testing the x509 APIs than the signers method.